src/app/runtime/runtime.wac.ts

```diff
@@ -54,85 +50,40 @@ export class WACRuntime implements Runtime {
-
-            if (!MsftSme.isNullOrUndefined(self.Resources.accessibilityMode) && !MsftSme.isNullOrUndefined(data.accessibilityMode)) {
-                self.Resources.accessibilityMode = data.accessibilityMode;
-                CoreEnvironment.accessibilityManager.changeAccessibilityMode(self.Resources.accessibilityMode);
-            }
-
-            // ensure that global environment settings persist across origins
-            if (!MsftSme.isNullOrUndefined(data.consoleDebug) && MsftSme.consoleDebug() !== data.consoleDebug) {
-                MsftSme.consoleDebug(data.consoleDebug);
-            }
-
-            if (!MsftSme.isNullOrUndefined(data.sideLoad)) {
-                let sideLoadKeys = Object.keys(MsftSme.sideLoad() || {});
-                let dataSideLoadKeys = Object.keys(data.sideLoad || {});
-
-                if (sideLoadKeys.length !== dataSideLoadKeys.length || sideLoadKeys.some(key => !!data.sideLoad[key])) {
-                    MsftSme.sideLoadReset();
-                    dataSideLoadKeys.forEach(k => MsftSme.sideLoad(k));
-                }
-            }
-
-            if (!MsftSme.isNullOrUndefined(data.experiments)) {
-                let experiments = MsftSme.experiments();
-                if (experiments.length !== data.experiments.length || experiments.some((v, i) => v !== data.experiments[i])) {
-                    MsftSme.experiments(data.experiments);
-                }
-            }
+    public OnModuleCreate(): void {
+        this.appContext.initializeModule({});
+    }
 
-            // Skipping loading css
-            // CoreEnvironment.assetManager.loadAssets(data.theme, data.assets);
-            const localeLoader = CoreEnvironment.moduleLoadLocale({ id: data.localeRegional || data.locale, neutral: data.locale });
-            // let passingData = <RpcInitData>{
-            //     locale: data.locale,
-            //     localeRegional: data.localeRegional,
-            //     sessionId: data.sessionId
-            // };
-            return localeLoader
-                .then(() => rpc.seekShell(RpcSeekMode.Create))
-                // Note that the handler callback simply set navigationService.active to true, which should already been the case
-                // .then(() => handler(passingData))
-                .then(() => { return { version: rpcVersion }; });
-        })
+    public OnAppInit(): void {
+        this.appContext.ngInit({ navigationService: this.navigationService });
     }
 
-    public DestroyContext() {
+    public OnAppDestroy(): void {
         if (this._tokenId) {
-            this.powershellService.run(PowerShellScripts.token_utils, {
+            this.powershellService.run(PowerShellScripts.token_utils.script, {
                 command: 'delete',
                 tokenId: this._tokenId,
                 apiHost: this._apiHost,
-            }).finally(() => this.appContext.ngDestroy()).subscribe()
+            }).subscribe(null, null, () => this.appContext.ngDestroy());

```

Comment:
Unsubscribe was not called in the complete callback function. Please add it if required.

Reply:
Note that this code is OnAppDestroy. All callbacks would go out of scope right after this method. We shouldn't need to do it. But I will add it just to be consistent